### PR TITLE
sd-bus: initialize return values on success in sd_bus_message_read_ar…

### PR DESCRIPTION
**Pull request by poettering**
…ray()

Fixes: #17346

### CODE REVIEW

**Comment by poettering** [comment]
Added a man page update, that clarifies return value of 0 vs. 1.

File: man/sd_bus_message_read_array.xml [hunk 1]
```diff
@@ -58,9 +58,12 @@
     <title>Return Value</title>
 
     <para>
-      On success, <function>sd_bus_message_read_array()</function> returns 0 or
-      a positive integer. On failure, it returns a negative errno-style error
-      code.
+      On success and when an array was read, <function>sd_bus_message_read_array()</function> returns a
+      positive integer greater than zero. If invoked while inside a container element (such as an array,
```
Comment (yuwata) [inline man/sd_bus_message_read_array.xml]: 'greater than zero' sounds redundant, as 'positive' implies that.

File: man/sd_bus_message_read_array.xml [hunk 2]
```diff
@@ -58,9 +58,12 @@
...
+      e.g. when operating on an array of arrays) and the final alement of the outer container has been read
```
Comment (yuwata) [inline man/sd_bus_message_read_array.xml]: alement -> element.

**Comment by poettering** [comment]
Force pushed new version. Only changes are the two @yuwata found. Taking liberty to upgrade green label, since so minor.

**Pull request merged**